src/Tests/Cache/ShardedPhpRedisUnitTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * Definition of Drupal\redis\Tests\Cache\PhpRedisUnitTest.
+ */
+
+namespace Drupal\redis\Tests\Cache;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Site\Settings;
+use Drupal\redis\Cache\ShardedPhpRedis;
+use Drupal\system\Tests\Cache\GenericCacheBackendUnitTestBase;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Tests PhpRedis cache backend using GenericCacheBackendUnitTestBase.
+ *
+ * @group redis
+ */
+class ShardedPhpRedisUnitTest extends GenericCacheBackendUnitTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = array('system', 'redis');
+
+  public function containerBuild(ContainerBuilder $container) {
+    parent::containerBuild($container);
+    // Replace the default checksum service with the redis implementation.
+    if ($container->has('redis.factory')) {
+      $container->register('cache_tags.invalidator.checksum', 'Drupal\redis\Cache\RedisCacheTagsChecksum')
+        ->addArgument(new Reference('redis.factory'))
+        ->addTag('cache_tags_invalidator');
+    }
+  }
+
+  /**
+   * Creates a new instance of PhpRedis cache backend.
+   *
+   * @return \Drupal\redis\Cache\PhpRedis
+   *   A new PhpRedis cache backend.
+   */
+  protected function createCacheBackend($bin) {
+    $cache = new ShardedPhpRedis(
+        $bin,
+        \Drupal::service('redis.factory')->getClient(),
+        \Drupal::service('cache_tags.invalidator.checksum')
+    );
+    $cache->setMinTtl(10);
+    return $cache;
+  }
+
+  /**
+   * Tests Drupal\Core\Cache\CacheBackendInterface::invalidateTags().
+   */
+  function testInvalidateTags() {
```

Comment:
Since that my shared backend cannot guarantee by its algorithm that we can keep invalidated entries in all cases, I had to override the follow test cases. In all cases I at worst just removed a single line. The invalid entry keeping is an optional feature and not all backend will be able to do it as soon as you expect the backend itself to deal with expiration. I had to ugly patch my cache backend to force it to keep entries in some cases, and I'm not happy with this, because core test class forces this feature to be mandatory for all backends using it...